library/packages/src/modules/Package.rb

```diff
@@ -221,6 +222,18 @@ def reset
       @removed_packages.clear
     end
 
+    # Tries to find a package according to the pattern
+    #
+    # @param pattern [String] a pattern to match, no escaping done
+    # @return list of matching package names
+    def by_pattern(pattern = "")
```

Comment:
providing a default value for `by_pattern` IMHO only encourages unreadable call sites